lib/resolution/ResolutionContext.ts

```diff
@@ -140,7 +140,14 @@ export class ResolutionContext {
 
         // If so, require the Node.js types, and resolve the file for the built-in package
         const rootFile = this.resolvePackageIndexInner(require, `@types/node`, currentFilePath);
-        return rootFile.replace(/index\.d\.ts$/u, `${packageName}.d.ts`);
+        const result = rootFile.replace(/index\.d\.ts$/u, `${packageName}.d.ts`);
+        // eslint-disable-next-line no-sync
+        if (!fs.existsSync(result)) {
```

Comment:
Currently, `resolvePackageIndex` is a fast sync method, but this can make it quite slow.
Can you investigate if there is another way forward?

Reply:
Since the method also calls `require.resolve`, I doubt we pay a big additional cost?

Reply:
If you are still worried about this, I can look at adding some cache (since this code is only triggered for node modules, the cache size should not be to big). - I am not committed to finding a vastly different approach sorry.